src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
+    {
+        VkImageCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // there are a lot of extensions
+        vk_createInfo.flags = static_cast<VkImageCreateFlags>(params.flags);
+        vk_createInfo.imageType = static_cast<VkImageType>(params.type);
+        vk_createInfo.format = getVkFormatFromFormat(params.format);
+        vk_createInfo.extent = { params.extent.width, params.extent.height, params.extent.depth };
+        vk_createInfo.mipLevels = params.mipLevels;
+        vk_createInfo.arrayLayers = params.arrayLayers;
+        vk_createInfo.samples = static_cast<VkSampleCountFlagBits>(params.samples);
+        vk_createInfo.tiling = static_cast<VkImageTiling>(params.tiling);
+        vk_createInfo.usage = static_cast<VkImageUsageFlags>(params.usage);
+        vk_createInfo.sharingMode = static_cast<VkSharingMode>(params.sharingMode);
+        vk_createInfo.queueFamilyIndexCount = params.queueFamilyIndices->size();
+        vk_createInfo.pQueueFamilyIndices = params.queueFamilyIndices->data();
+        vk_createInfo.initialLayout = static_cast<VkImageLayout>(params.initialLayout);
+
+        VkImage vk_image;
+        if (vkCreateImage(m_vkdev, &vk_createInfo, nullptr, &vk_image) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanImage>(core::smart_refctd_ptr<CVKLogicalDevice>(this),
+                std::move(params), vk_image);
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+    bool bindImageMemory(uint32_t bindInfoCount, const SBindImageMemoryInfo* pBindInfos) override
+    {
+        for (uint32_t i = 0u; i < bindInfoCount; ++i)
+        {
+            if (pBindInfos[i].image->getAPIType() != EAT_VULKAN)
+                continue;
+            VkImage vk_image = static_cast<const CVulkanImage*>(pBindInfos[i].image)->getInternalObject();
+
+            // if (pBindInfos[i].memory->getAPIType() != EAT_VULKAN)
+            //     continue;
+            VkDeviceMemory vk_deviceMemory = static_cast<const CVulkanMemoryAllocation*>(pBindInfos[i].memory)->getInternalObject();
+
+            if (vkBindImageMemory(m_vkdev, vk_image, vk_deviceMemory,
+                static_cast<VkDeviceSize>(pBindInfos[i].offset)) != VK_SUCCESS)
+            {
+                return false;
+            }
+        }
+
+        return true;
     }
             
     core::smart_refctd_ptr<IGPUImage> createGPUImageOnDedMem(IGPUImage::SCreationParams&& params, const IDriverMemoryBacked::SDriverMemoryRequirements& initialMreqs) override
     {
+        // Todo(achal)
+#if 0
+        core::smart_refctd_ptr<IGPUImage> gpuImage = createGPUImage(core::smart_refctd_ptr(params));
+        if (!gpuImage)
+            return nullptr;
+        
+        core::smart_refctd_ptr<video::IDriverMemoryAllocation> imageMemory =
+            allocateDeviceLocalMemory(gpuImage->getMemoryReqs());
+#endif
+
```

Comment:
same spiel as creating GPU Buffer on ded mem:

1. create Image
2. query memory requirements
Common utility for all 10+ functions:
3. merge the intiailMemreqs with the queries memory reqs
4. choose memory type/heap
5. allocate
Then come back
6. bind memory

Reply:
Resolved

Reply:
obsolete!